x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/MaybeViewTraceLink.tsx

```diff
@@ -73,6 +75,11 @@ export const MaybeViewTraceLink = ({
 
     // the user is viewing a zoomed in version of the trace. Link to the full trace
   } else {
+    const nextEnvironment = getEnvironmentUrlParam(
+      rootTransaction?.service.environment,
+      environment
+    );
```

Comment:
This would be easier to understand if it was destructured:

```suggestion
    const nextEnvironment = getEnvironmentUrlParam({
      requestedEnvironment: rootTransaction?.service.environment,
      currentEnvironment: environment
    });
```